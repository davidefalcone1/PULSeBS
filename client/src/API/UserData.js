class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

    static fromJson(json){
        const temp =  Object.assign(new UserData(), json);
        return temp;
    }
}

export default UserData;