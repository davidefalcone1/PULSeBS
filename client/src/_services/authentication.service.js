import { BehaviorSubject } from 'rxjs';


const currentUserSubject = new BehaviorSubject(JSON.parse(localStorage.getItem('user')));

export const authenticationService = {
    login,
    logout,
    user: currentUserSubject.asObservable(),
    get currentUserValue () { return currentUserSubject.value }
};

function handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            if ([401, 403].indexOf(response.status) !== -1) {
                // auto logout if 401 Unauthorized or 403 Forbidden response returned from api
                authenticationService.logout();
            }

            const error = (data && data.message) || response.statusText;
            return Promise.reject(error);
        }

        return data;
    });
}

function login(username, password) {
    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ username, password })
    };
    return new Promise(async function (resolve, reject) {
        try {
            const response = await fetch('/users/authenticate', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ username, password }),
            });
            const user = await response.json().then(handleResponse);
            if (response.ok) {
                localStorage.setItem('user', JSON.stringify(user));
                currentUserSubject.next(user);
                resolve(user);
            }
            else
                reject();
        }
        catch (e) {
            reject();
        }
    });

}
function logout() {

    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
    };
    localStorage.removeItem('user');
    currentUserSubject.next(null);
    return fetch('/logout',requestOptions);
}