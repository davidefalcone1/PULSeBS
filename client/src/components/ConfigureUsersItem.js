import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import Button from 'react-bootstrap/Button';

const configureUserItem = (props) => {

  return (
    <ListGroup.Item id = {"user-" + props.user.person}>
        <div className="d-flex w-100 pt-3 justify-content-between no-gutters">
            <IdField id = {props.user.personId}/>
            <NameField id = {props.user.personId} name = {props.user.fullName}/>
            <EmailField id = {props.user.personId} email = {props.user.email}/>
            <ButtonField id = {props.user.personId} editUser = {props.editUser}/>
        </div>
    </ListGroup.Item>
  );
}

function IdField(props){
    return(
        <div className="col-sm-3">
            <h6 id={"idOfuser_" + props.id}>
                {props.id}
            </h6>
        </div>
    );
}

function NameField(props){
    return(
        <div className="col-sm-3">
            <h6 id={"nameOfuser_" + props.id}>
                {props.name}
            </h6>
        </div>
    );
}

function EmailField(props){
    return(
        <div className="col-sm-4">
            <h6 id={"emailOfuser_" + props.id}>
                {props.email}
            </h6>
        </div>
    );
}


function ButtonField(props){
    return(
        <div className="col-sm-2">
            <Button variant="primary" onClick={(event) => {
                event.preventDefault();
                props.editUser(); //TODO SEND PARAMETERS
            }} id={"buttonFieldOfuser" + props.id}>
                Edit
            </Button>
        </div>
    );
}

export default configureUserItem;