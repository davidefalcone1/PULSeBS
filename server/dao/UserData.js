class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

}

module.exports = UserData;