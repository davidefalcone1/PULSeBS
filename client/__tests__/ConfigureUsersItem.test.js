import React from 'react';
import renderer from 'react-test-renderer';
import { AuthContext } from '../src/_services/AuthContext';
import ConfigureUsersItem from '../src/components/ConfigureUsersItem';
import { BrowserRouter } from 'react-router-dom';
import UserData from '../src/API/UserData';

describe('ConfigureUsersItem', ()=>{
    test('Renders correctly', ()=>{
        const component = renderer.create(
            <BrowserRouter>
                <AuthContext.Provider value={{user: new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')}}>
                    <ConfigureUsersItem user={new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')}/>
                </AuthContext.Provider>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});