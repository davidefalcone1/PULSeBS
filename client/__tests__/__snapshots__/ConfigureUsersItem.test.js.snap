// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureUsersItem Renders correctly 1`] = `
<div
  className="list-group-item"
  disabled={false}
  id="user-undefined"
  onClick={[Function]}
>
  <div
    className="d-flex w-100 pt-3 justify-content-between no-gutters"
  >
    <div
      className="col-sm-3"
    >
      <h6
        id="idOfuser_12345"
      >
        12345
      </h6>
    </div>
    <div
      className="col-sm-4"
    >
      <h6
        id="nameOfuser_12345"
      >
        Davide Falcone
      </h6>
    </div>
    <div
      className="col-sm-5"
    >
      <h6
        id="emailOfuser_12345"
      >
        davide.falcone@polito.it
      </h6>
    </div>
  </div>
</div>
`;
